transmission_interface/include/transmission_interface/simple_transmission.h

```diff
@@ -0,0 +1,210 @@
+ *   radii for the timing belt; or the ratio between output and input teeth for the gear system.
+ *   The transmission ratio can take any real value \e except zero. In particular:
+ *     - If its absolute value is greater than one, it's a velocity reducer / effort amplifier, while if its absolute
+ *       value lies in \f$ (0, 1) \f$ it's a velocity amplifier / effort reducer.
+ *     - Negative values represent a direction flip, ie. actuator and joint move in opposite directions. For example,
+ *       in timing belts actuator and joint move in the same direction, while in single-stage gear systems actuator and
+ *       joint move in opposite directions.
+ *
+ * \ingroup transmission_types
+ */
+class SimpleTransmission: public Transmission
+{
+public:
+    /**
+     * \param reduction Reduction ratio.
+     * \param joint_offset Joint position offset used in the position mappings.
+     * \pre Nonzero reduction value.
+     */
+    SimpleTransmission(
+      const double reduction,
+      const double joint_offset = 0.0);
+
+    void configure(
+      const std::vector < hardware_interface::CommandInterface > & joint_handles,
+      const std::vector < hardware_interface::CommandInterface > & actuator_handles) override;
+
+    /**
+     * \brief Transform \e effort variables from actuator to joint space.
+     * \param[in]  act_data Actuator-space variables.
+     * \param[out] jnt_data Joint-space variables.
+     * \pre Actuator and joint effort vectors must have size 1 and point to valid data.
+     *  To call this method it is not required that all other data vectors contain valid data, and can even remain empty.
+     */
+    void actuator_to_joint() override;
+
+    /**
+     * \brief Transform \e effort variables from joint to actuator space.
+     * \param[in]  jnt_data Joint-space variables.
+     * \param[out] act_data Actuator-space variables.
+     * \pre Actuator and joint effort vectors must have size 1 and point to valid data.
+     *  To call this method it is not required that all other data vectors contain valid data, and can even remain empty.
+     */
+    void joint_to_actuator() override;
+
+    std::size_t num_actuators() const override {return 1;}
+    std::size_t num_joints()    const override {return 1;}
+
+    double get_actuator_reduction() const {return reduction_;}
+    double get_joint_offset()       const {return jnt_offset_;}
+
+protected:
+    double reduction_;
+    double jnt_offset_;
+
+    hardware_interface::CommandInterface joint_position = {"", "", nullptr};
+    hardware_interface::CommandInterface joint_velocity = {"", "", nullptr};
+    hardware_interface::CommandInterface joint_effort = {"", "", nullptr};
+
+    hardware_interface::CommandInterface actuator_position = {"", "", nullptr};
+    hardware_interface::CommandInterface actuator_velocity = {"", "", nullptr};
+    hardware_interface::CommandInterface actuator_effort = {"", "", nullptr};
```

Comment:
I don't think that's the right approach. The actual command interfaces shall always be owned by the resource manager and should only be loaned to the instances. Why does a transmission have to own the command interfaces?

Reply:
I'm now using joint and actuator handles which are internal types of the transmission_interface library. The library is meant to be used internally in components to implement transmissions, there is no resourcemanager and loaning in this context, only internal business.

Reply:
Please see the updated version

Reply:
did you forgot to push a change? I don't see any updates here.